ral-facilities/operationsgateway
# DSEGOG-59 DSEGOG-61 plot colours

## Description
Joint PR from me and @louise-davies

This allows the user to change the colour of scatter points and lines for a plotted channel on the graph in real time. This uses the [react-colorful](https://www.npmjs.com/package/react-colorful) library's HexColorPicker to let the user choose any colour they want for a channel.

In addition to this, I have also implemented the functionality from eCat which automatically assigns a colour to a channel when it is selected to be plotted. It assigns up to ten specific colours for ten separate channels before assigning a random colour for any channels after this (these specific colours are the first ten from eCat, not including a grey one which wouldn't have looked great on a white background). 

## Testing instructions
Try choosing colours! Also check that colours are automatically assigned correctly (in the order dark green, dark blue, pink, light blue, etc.)


## Agile board tracking
Closes DSEGOG-59, DSEGOG-61
